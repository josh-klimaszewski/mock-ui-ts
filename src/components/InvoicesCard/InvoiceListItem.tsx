import { ListItem, ListItemText } from '@material-ui/core';
import { ChevronRight } from '@material-ui/icons';
import * as React from 'react';

const InvoiceListItem = ({ invoice }) => (
    <ListItem>
        <ListItemText primary={`${invoice.fields.customerId}`}/>
        ${invoice.fields.amount}
        <ChevronRight />
    </ListItem>
)
export default InvoiceListItem;