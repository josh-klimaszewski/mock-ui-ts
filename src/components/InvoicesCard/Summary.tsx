import { Collapse, ListItem, ListItemText } from '@material-ui/core';
import { ExpandLess, ExpandMore } from '@material-ui/icons';
import * as React from 'react';


class Summary extends React.Component<any, any> {
    constructor(props: any) {
        super(props);
        this.state = {
            open: false
        }
    }
    public handleToggle = () => {
        this.setState(({ open }) => ({
            open: !open
        }));
    };
    public render() {
        const { primary, amount, renderExpanded } = this.props;
        return(
            <React.Fragment>
                <ListItem className="list" onClick={this.handleToggle}>
                    <div className="list-item">
                        <ListItemText 
                            primary={primary}
                        />
                        ${amount}
                        {this.state.open ? <ExpandLess /> : <ExpandMore />}
                    </div>
                    <Collapse in={this.state.open} timeout="auto" unmountOnExit={true}>
                        {renderExpanded && renderExpanded()}
                    </Collapse>
                </ListItem>
            </React.Fragment>
        )
    }
}
export default Summary;
