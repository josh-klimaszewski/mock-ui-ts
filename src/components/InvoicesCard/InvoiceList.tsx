
import { List } from '@material-ui/core';
import * as React from 'react';
import InvoiceListItem from './InvoiceListItem';


const InvoiceList = ({ invoices }) => (
    <List>
        {invoices.map((invoice) => (
            <InvoiceListItem invoice={invoice} key={invoice.id} />
        ))}
    </List>
    
);
export default InvoiceList;