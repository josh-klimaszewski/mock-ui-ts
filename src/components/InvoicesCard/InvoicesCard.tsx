import { Card, CardContent, Divider, List } from '@material-ui/core';
import * as React from "react";
import { connect } from "react-redux";
import { fetchInvoices } from "../../actions";

import InvoiceList from './InvoiceList';
import './style.css';
import Summary from './Summary';



// import { Card, CardContent, Divider, List } from '@material-ui/core';
export class InvoicesCard extends React.Component<any, any> {
  constructor(props: any) {
    super(props);
    this.state = {
        invoiceTotal: 0,
        invoices: []
    };
    this.getInvoiceTotal = this.getInvoiceTotal.bind(this);
  }
  public getInvoiceTotal = () => {
    const invoices = this.props.invoices;
    return invoices.reduce(
      (total: any, current: any) => total + current.fields.amount,
      0
    );
    
  };
  public componentDidMount() {
    this.props.fetchInvoices();
  }
  public render() {
    const invoiceTotal = this.getInvoiceTotal();
    return (
        <Card className="card">
            <CardContent className="card-header">
                Invoices
            </CardContent>
            <List>
                <Summary 
                    className="summary"
                    primary="Coming Due"
                    amount={invoiceTotal}
                    renderExpanded={() => <InvoiceList invoices={this.props.invoices} />}
                />
                <Divider />
                <Summary 
                    className="summary"
                    primary="Past Due"
                    amount={5560}
                    renderExpanded={() => <InvoiceList invoices={this.props.invoices} />}
                />
                <Divider />
            </List>
        </Card>
    )
  }
}

function mapStateToProps(state: any) {
  return { invoices: state.invoices.invoices };
}
export default connect(
  mapStateToProps,
  { fetchInvoices }
)(InvoicesCard);
