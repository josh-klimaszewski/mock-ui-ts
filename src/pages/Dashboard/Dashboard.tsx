import * as React from 'react';
import InvoicesCard from '../../components/InvoicesCard'
import './style.css'

const Dashboard = () => {
    return (
        <div><InvoicesCard /></div>
    );
}
export default Dashboard;