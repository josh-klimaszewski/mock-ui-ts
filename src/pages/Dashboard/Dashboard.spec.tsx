import * as enzyme from 'enzyme';
import * as React from 'react';

import InvoicesCard from '../../components/InvoicesCard';
import Dashboard from './Dashboard';

describe('Dashboard component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = enzyme.shallow(<Dashboard />)
    })
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('renders InvoicesCard component', () => {
        expect(wrapper.contains(<InvoicesCard />)).toBe(true);
    })
});




