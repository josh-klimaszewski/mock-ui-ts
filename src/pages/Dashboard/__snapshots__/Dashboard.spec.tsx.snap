// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="feed"
        >
          <div
            className="cards"
          >
            <Connect(InvoicesCard) />
            <Connect(InvoicesCard) />
          </div>
        </div>,
        <div
          className="summary"
        >
          <div
            className="summaryCard"
          >
            <div
              className="totalCashWrapper"
            >
              <h6>
                Total Cash
              </h6>
              <h5>
                $11,200
              </h5>
            </div>
          </div>
        </div>,
      ],
      "className": "dashboard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="cards"
          >
            <Connect(InvoicesCard) />
            <Connect(InvoicesCard) />
          </div>,
          "className": "feed",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(InvoicesCard) />,
              <Connect(InvoicesCard) />,
            ],
            "className": "cards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="summaryCard"
          >
            <div
              className="totalCashWrapper"
            >
              <h6>
                Total Cash
              </h6>
              <h5>
                $11,200
              </h5>
            </div>
          </div>,
          "className": "summary",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="totalCashWrapper"
            >
              <h6>
                Total Cash
              </h6>
              <h5>
                $11,200
              </h5>
            </div>,
            "className": "summaryCard",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h6>
                  Total Cash
                </h6>,
                <h5>
                  $11,200
                </h5>,
              ],
              "className": "totalCashWrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Total Cash",
                },
                "ref": null,
                "rendered": "Total Cash",
                "type": "h6",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$11,200",
                },
                "ref": null,
                "rendered": "$11,200",
                "type": "h5",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="feed"
          >
            <div
              className="cards"
            >
              <Connect(InvoicesCard) />
              <Connect(InvoicesCard) />
            </div>
          </div>,
          <div
            className="summary"
          >
            <div
              className="summaryCard"
            >
              <div
                className="totalCashWrapper"
              >
                <h6>
                  Total Cash
                </h6>
                <h5>
                  $11,200
                </h5>
              </div>
            </div>
          </div>,
        ],
        "className": "dashboard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="cards"
            >
              <Connect(InvoicesCard) />
              <Connect(InvoicesCard) />
            </div>,
            "className": "feed",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(InvoicesCard) />,
                <Connect(InvoicesCard) />,
              ],
              "className": "cards",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="summaryCard"
            >
              <div
                className="totalCashWrapper"
              >
                <h6>
                  Total Cash
                </h6>
                <h5>
                  $11,200
                </h5>
              </div>
            </div>,
            "className": "summary",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="totalCashWrapper"
              >
                <h6>
                  Total Cash
                </h6>
                <h5>
                  $11,200
                </h5>
              </div>,
              "className": "summaryCard",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h6>
                    Total Cash
                  </h6>,
                  <h5>
                    $11,200
                  </h5>,
                ],
                "className": "totalCashWrapper",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Total Cash",
                  },
                  "ref": null,
                  "rendered": "Total Cash",
                  "type": "h6",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "$11,200",
                  },
                  "ref": null,
                  "rendered": "$11,200",
                  "type": "h5",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
